mlflow/utils/search_utils.py

```diff
@@ -26,8 +26,12 @@ class SearchUtils(object):
         CASE_INSENSITIVE_STRING_COMPARISON_OPERATORS.union({'='})
     VALID_SEARCH_ATTRIBUTE_KEYS = set(RunInfo.get_searchable_attributes())
     VALID_ORDER_BY_ATTRIBUTE_KEYS = set(RunInfo.get_orderable_attributes())
-    VALID_ORDER_BY_KEYS_REGISTERED_MODELS = set([SqlRegisteredModel.name.key,
-                                                 SqlRegisteredModel.last_updated_time.key])
+    VALID_TIMESTAMP_ORDER_BY_KEYS = set(["timestamp", "last_updated_timestamp"])
+    VALID_ORDER_BY_KEYS_REGISTERED_MODELS = set([SqlRegisteredModel.name.key])\
+        .union(VALID_TIMESTAMP_ORDER_BY_KEYS)
+    # We encourage users to use timestamp for order-by
+    RECOMMENDED_ORDER_BY_KEYS_REGISTERED_MODELS = VALID_ORDER_BY_KEYS_REGISTERED_MODELS\
```

Comment:
- create a constant for `"timestamp"`
- replace `RECOMMENDED_ORDER_BY_KEYS_REGISTERED_MODELS` with something like this

```
DOCUMENTED_ORDER_BY_KEYS_FOR_REGISTERED_MODELS = [ORDER_BY_KEY_MODEL_NAME, ORDER_BY_KEY_TIMESTAMP]
```